cmd/kubeadm/app/phases/etcd/local.go

```diff
@@ -57,6 +57,7 @@ func GetEtcdPodSpec(cfg *kubeadmapi.MasterConfiguration) v1.Pod {
 		Command:         getEtcdCommand(cfg),
 		Image:           images.GetCoreImage(kubeadmconstants.Etcd, cfg.ImageRepository, cfg.KubernetesVersion, cfg.Etcd.Image),
 		ImagePullPolicy: cfg.ImagePullPolicy,
+		Env: []v1.EnvVar{{Name: "ETCDCTL_API", Value: "3"}},
```

Comment:
Well, I don't like this change.
This change is not indispensable to start an Etcd pod.

In your case, I'd prefer to rebuild an Etcd docker images and inject these environment variables.